go/ct/spc/specification.go

```diff
@@ -2256,48 +2262,72 @@ func nonStaticSelfDestructRules(revision tosca.Revision, warm bool, destinationC
 			Eq(ReadOnly(), false),
 			IsRevision(revision),
 			hasSelfDestructedCondition,
-			targetWarm,
+			beneficiaryExists,
+			beneficiaryWarm,
 		},
 		parameters: []Parameter{AddressParameter{}},
 		effect: func(s *st.State) {
-			selfDestructEffect(s, destinationColdCost, accountCreationFee, refundGas)
+			selfDestructEffect(s)
 		},
 	}
 
 	return rulesFor(instruction)
 }
 
-func selfDestructEffect(s *st.State, destinationColdCost, accountCreationFee, refundGas tosca.Gas) {
+func selfDestructEffect(s *st.State) {
 	// Behavior pre cancun: the current account is registered to be destroyed, and will be at the end of the current
 	// transaction. The transfer of the current balance to the given account cannot fail. In particular,
 	// the destination account code (if any) is not executed, or, if the account does not exist, the
 	// balance is still added to the given address.
 
-	// account to send the current balance to
-	destinationAccount := s.Stack.Pop().Bytes20be()
-	currentAccount := s.CallContext.AccountAddress
-	CurrentBalance := s.Accounts.GetBalance(currentAccount)
+	beneficiaryAccount := s.Stack.Pop().Bytes20be()
+	originatorAccount := s.CallContext.AccountAddress
+	originatorBalance := s.Accounts.GetBalance(originatorAccount)
 
 	dynamicCost := tosca.Gas(0)
-	if !CurrentBalance.IsZero() && !s.Accounts.Exist(destinationAccount) {
-		dynamicCost += accountCreationFee
-	}
 
-	dynamicCost += destinationColdCost
+	// Add costs for transfering the remaining balance.
+	if !originatorBalance.IsZero() {
+		// If the target account does not exist, the account creation fee is added.
+		if !s.Accounts.Exists(beneficiaryAccount) {
+			dynamicCost += 25000
+		}
+
+		// Add warm-up costs if the beneficiary account is cold.
+		if s.Revision > tosca.R07_Istanbul && !s.Accounts.IsWarm(beneficiaryAccount) {
```

Comment:
I know that this was written already like this, but I would prefer to have
```suggestion
		if s.Revision >= tosca.R08_Berlin && !s.Accounts.IsWarm(beneficiaryAccount) {
```
To facilitate recognition of the EIP-2929 semantics introduced in Berlin.